### DISCUSSION TITLE
How to Get Assets from Scheduled Job?

### DISCUSSION THREAD

**Discussion by dennisgera** [dagster-io/dagster#22588] [Q&A] [answered]
What's the best way to get the assets selected by a job in a schedule definition? I'm guessing the best way would be to use the schedule's context, but could not find documentation on how to do so. Here's an example of what I'm doing and what I'd like to do:

What I'm doing:
```python
@schedule(
    name="load_master_data_schedule",
    cron_schedule="0 6 * * *",
    default_status=default_status.schedule,
    job=load_master_data,
    execution_timezone=TZ,
)
def load_master_data_schedule(context: ScheduleEvaluationContext):
    assets = ["organizations", "clients", "b2b_users", "cost_centers", "organization_requests"]
    asset_keys = [AssetKey(["raw", "raw_masterdata", asset]) for asset in assets]

    return create_master_data_run_request(context, asset_keys)
```

What I'd like to do:
```python
@schedule(
    name="load_master_data_schedule",
    cron_schedule="0 6 * * *",
    default_status=default_status.schedule,
    job=load_master_data,
    execution_timezone=TZ,
)
def load_master_data_schedule(context: ScheduleEvaluationContext):
    assets = context.something
    asset_keys = [AssetKey(["raw", "raw_masterdata", asset]) for asset in assets]

    return create_master_data_run_request(context, asset_keys)
```

Is this possible?

**Comment by garethbrickman** [comment, marked as answer]
After doing some testing I discovered that `JobDefinition` has an attribute called [_all_node_defs](https://docs.dagster.io/_modules/dagster/_core/definitions/job_definition#JobDefinition:~:text=_all_node_defs%3A%20Mapping%5Bstr%2C%20NodeDefinition%5D) which returns a dictionary whose keys are the names of the assets used in the job.

```python
# Access the job definition from the repository
job_def = context.repository_def.get_job("asset_job")
all_node_defs = job_def._all_node_defs

# Extract and log the keys which are the asset names
for asset_name in all_node_defs.keys():
    context.log.info(f"Asset: {asset_name}")
```

There's also an attribute called [_asset_layer](https://docs.dagster.io/_modules/dagster/_core/definitions/job_definition#JobDefinition:~:text=_asset_layer%3A%20AssetLayer) that contains lots of info, including the job's asset keys.
